diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92c0e34..f0bd3eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -545,9 +545,9 @@ if (Unwind_FOUND)
   set (Unwind_DEPENDENCY "find_dependency (Unwind ${Unwind_VERSION})")
 endif (Unwind_FOUND)
 
-if (HAVE_DBGHELP)
+if (HAVE_DBGHELP OR MINGW)
    target_link_libraries (glog PUBLIC dbghelp)
-endif (HAVE_DBGHELP)
+endif (HAVE_DBGHELP OR MINGW)
 
 if (HAVE_PTHREAD)
   target_link_libraries (glog PUBLIC ${CMAKE_THREAD_LIBS_INIT})
diff --git a/src/logging.cc b/src/logging.cc
index 42cfd01..2af707a 100644
--- a/src/logging.cc
+++ b/src/logging.cc
@@ -1209,7 +1209,7 @@ void LogFileObject::Write(bool force_flush,
                        << setw(2) << tm_time.tm_sec << '\n'
                        << "Running on machine: "
                        << LogDestination::hostname() << '\n'
-                       << "Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu "
+                       << "Log line format: [IWEF]mmdd hh:mm:ss "
                        << "threadid file:line] msg" << '\n';
     const string& file_header_string = file_header_stream.str();
 
@@ -1431,8 +1431,7 @@ void LogMessage::Init(const char* file,
              << ' '
              << setw(2) << data_->tm_time_.tm_hour  << ':'
              << setw(2) << data_->tm_time_.tm_min   << ':'
-             << setw(2) << data_->tm_time_.tm_sec   << "."
-             << setw(6) << data_->usecs_
+             << setw(2) << data_->tm_time_.tm_sec
              << ' '
              << setfill(' ') << setw(5)
              << static_cast<unsigned int>(GetTID()) << setfill('0')
@@ -1832,8 +1831,7 @@ string LogSink::ToString(LogSeverity severity, const char* file, int line,
          << ' '
          << setw(2) << tm_time->tm_hour << ':'
          << setw(2) << tm_time->tm_min << ':'
-         << setw(2) << tm_time->tm_sec << '.'
-         << setw(6) << usecs
+         << setw(2) << tm_time->tm_sec
          << ' '
          << setfill(' ') << setw(5) << GetTID() << setfill('0')
          << ' '
diff --git a/src/raw_logging.cc b/src/raw_logging.cc
index 3bbfda3..d0d3ec4 100644
--- a/src/raw_logging.cc
+++ b/src/raw_logging.cc
@@ -114,7 +114,7 @@ void RawLog__(LogSeverity severity, const char* file, int line,
   int size = sizeof(buffer);
 
   // NOTE: this format should match the specification in base/logging.h
-  DoRawLog(&buf, &size, "%c0000 00:00:00.000000 %5u %s:%d] RAW: ",
+  DoRawLog(&buf, &size, "%c0000 00:00:00 %5u %s:%d] RAW: ",
            LogSeverityNames[severity][0],
            static_cast<unsigned int>(GetTID()),
            const_basename(const_cast<char *>(file)), line);
